lib/src/auth/frappe/frappe.auth.controller.dart

```diff
@@ -763,6 +763,22 @@ class FrappeAuthController extends AuthController<FrappeSessionStatusInfo> {
         : RequestResponse.fail(response.error);
   }
 
+  /// Sets the session locally obtained externally and validates against the backend.
+  ///
+  /// Useful when a custom API returns a session object, "Sign Up", for instance.
+  ///
+  /// The session must be valid. In case the session is to be cleared, use [logout].
+  @override
+  Future<RequestResponse<FrappeSessionStatusInfo>> setExternalSession(
+      FrappeSessionStatusInfo sessionStatusInfo) {
+    assert(sessionStatusInfo != null);
+    assert(sessionStatusInfo.user != null,
+        'Only a valid session can be set.\nUse .logout() if you want to clear the session');
+
+    updateSession(sessionStatus: sessionStatusInfo, loggedIn: true);
```

Comment:
I'm not quite sure if the `loggedIn` should be left as `true`. Any context to why it shouldn't be passed as a param to this function?

Reply:
Because this method should only be used when you actually have a session. There's no point in using this method if no session or an invalid one is returned by a custom API.